src/main/java/lotto/domain/result/GameResult.java

```diff
@@ -13,7 +13,7 @@ public class GameResult {
   private final Map<Rank, Integer> details;
   private final BigDecimal rate;
 
-  public GameResult(Map<Rank, Integer> details, Money winningPrize, Money purchaseAmount){
+  public GameResult(Map<Rank, Integer> details, Money winningPrize, Money purchaseAmount) {
```

Comment:
```suggestion
  public GameResult(Map<Rank, Integer> details) {
```
앞에서도 피드백한 바와 같이 GameResult는 등수 결과만 저장하고 관리하는 Map의 일급 콜렉션으로 구현하면 어떨까?
Map 정보로 우승 상금을 구할 수 있고, 수익률 계산을 위한 구매금액은 메서드 인자로 전달해도 되지 않을까?